Migrate from F8 to Dekorate

Sgitario opened the pull request:
Instead of using fabric8 maven plugin to generate the resources, we'll now use dekorate and the dekorate test framework to run the test in a managed approach and unmanaged approach (deploying the app using s2i before running the test).

jacobdotcosta reviewed (changes requested):
Could you include in this PR updating the README.md instructions?

Deploying keycloak on Openshift is done using `oc create -f .openshiftio/sso.yaml` instead of `oc create -f service.sso.yaml`.

Could you also include the instructions to run the unit tests, as you did for instance in the `cache-example`?

Sgitario committed:
Migrate from F8 to Dekorate

Instead of using fabric8 maven plugin to generate the resources, we'll now use dekorate and the dekorate test framework to run the test in a managed approach and unmanaged approach (deploying the app using s2i before running the test).

jacobdotcosta reviewed (approved):
LGTM
